dotcom-rendering/src/web/components/TopRightAdSlot.importable.tsx

```diff
@@ -37,50 +39,74 @@ export const TopRightAdSlot = ({
 	format?: ArticleFormat;
 }) => {
 	const adBlockerDetected = useAdBlockInUse();
-	const isSignedIn =
-		!isServer && !!getCookie({ name: 'GU_U', shouldMemoize: true });
+	const [isShady, setIsShady] = useState(false);
 
-	if (
-		adBlockerDetected &&
-		!isSignedIn &&
-		!shouldHideReaderRevenue &&
-		!isPaidContent &&
-		!isServer
-	) {
-		return <ShadyPie format={format} editionId={editionId} />;
-	}
+	const ABTestAPI = useAB();
+
+	const userInShadyPieTestVariant =
+		ABTestAPI?.isUserInVariant('shadyPieClickThrough', 'variant') &&
+		format?.theme == ArticlePillar.Lifestyle;
 
-	// Otherwise return the classic ad slot
-	return (
-		<div
-			id="top-right-ad-slot"
-			css={css`
-				position: static;
-				height: ${MOSTVIEWED_STICKY_HEIGHT}px;
-			`}
-		>
+	useEffect(() => {
+		const isSignedIn =
+			!isServer && !!getCookie({ name: 'GU_U', shouldMemoize: true });
+		setIsShady(
+			(adBlockerDetected &&
+				!isSignedIn &&
+				!shouldHideReaderRevenue &&
+				!isPaidContent &&
+				!isServer) ||
+				!!userInShadyPieTestVariant,
```

Comment:
I think you can entirely delete all references to `isServer` - it's only ever used inside the body of the `useEffect` and `useEffect` doesn't run on the server, so it'd always be `false`.